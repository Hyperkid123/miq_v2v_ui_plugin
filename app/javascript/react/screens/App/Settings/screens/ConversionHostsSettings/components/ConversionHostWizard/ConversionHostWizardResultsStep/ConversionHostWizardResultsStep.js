import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'patternfly-react';

class ConversionHostWizardResultsStep extends React.Component {
  componentDidMount() {
    const { postBodies, postConversionHostsAction, postConversionHostsUrl } = this.props;
    postConversionHostsAction(postConversionHostsUrl, postBodies);
  }

  render() {
    const { isPostingConversionHosts, isRejectedPostingConversionHosts, postConversionHostsResults } = this.props;
    return (
      <Form className="form-horizontal">
        <h2>TODO: Results Step Contents</h2>
        <ul>
          <li>Posting? {isPostingConversionHosts}</li>
          <li>Rejected? {isRejectedPostingConversionHosts}</li>
          <li>Results: {JSON.stringify(postConversionHostsResults, 4)}</li>
        </ul>
      </Form>
    );
  }
}

ConversionHostWizardResultsStep.propTypes = {
  postBodies: PropTypes.arrayOf(PropTypes.object),
  postConversionHostsAction: PropTypes.func,
  postConversionHostsUrl: PropTypes.string,
  isPostingConversionHosts: PropTypes.bool,
  isRejectedPostingConversionHosts: PropTypes.bool,
  postConversionHostsResults: PropTypes.arrayOf(PropTypes.object)
};

ConversionHostWizardResultsStep.defaultProps = {
  postConversionHostsUrl: '/api/conversion_hosts'
};

export default ConversionHostWizardResultsStep;
