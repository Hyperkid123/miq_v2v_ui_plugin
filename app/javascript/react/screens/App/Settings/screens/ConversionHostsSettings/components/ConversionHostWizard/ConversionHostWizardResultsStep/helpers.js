export const getConfigureConversionHostPostBody = (locationStepValues, hostsStepValues, authStepValues) => {
  console.log('TODO: use location step values:', locationStepValues);
  console.log('TODO: use hosts step values:', hostsStepValues);
  console.log('TODO: use auth step values: ', authStepValues);
  // TODO
  return {
    foo: 'bar'
  };
};
