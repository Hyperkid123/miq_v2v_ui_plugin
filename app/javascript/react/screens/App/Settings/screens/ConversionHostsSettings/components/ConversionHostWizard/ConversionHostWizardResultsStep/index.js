import { connect } from 'react-redux';
import ConversionHostWizardResultsStep from './ConversionHostWizardResultsStep';

import { stepIDs } from '../ConversionHostWizardConstants';
import { getConfigureConversionHostPostBody } from './helpers';

const mapStateToProps = ({ form }) => {
  const locationStepValues = form[stepIDs.locationStep] && form[stepIDs.locationStep].values;
  const hostsStepValues = form[stepIDs.hostsStep] && form[stepIDs.hostsStep].values;
  const authStepValues = form[stepIDs.authenticationStep] && form[stepIDs.authenticationStep].values;
  const postBody = getConfigureConversionHostPostBody(locationStepValues, hostsStepValues, authStepValues);
  return { postBody };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  {},
  mergeProps
)(ConversionHostWizardResultsStep);
