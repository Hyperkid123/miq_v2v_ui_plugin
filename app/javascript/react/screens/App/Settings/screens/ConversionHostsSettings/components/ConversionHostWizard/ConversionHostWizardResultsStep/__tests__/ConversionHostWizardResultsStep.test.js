import React from 'react';
import { shallow } from 'enzyme';

import ConversionHostWizardResultsStep from '../ConversionHostWizardResultsStep';

describe('conversion host wizard results step', () => {
  const getBaseProps = () => ({
    postBodies: [{ mock: 'post body' }],
    postConversionHostsAction: jest.fn(),
    postConversionHostsUrl: '/api/mock/conversion_hosts',
    isPostingConversionHosts: false,
    isRejectedPostingConversionHosts: false,
    postConversionHostsResults: null
  });

  it('calls postConversionHostsAction on mount', () => {
    const baseProps = getBaseProps();
    shallow(<ConversionHostWizardResultsStep {...baseProps} />);
    expect(baseProps.postConversionHostsAction).toHaveBeenCalledTimes(1);
    expect(baseProps.postConversionHostsAction).toHaveBeenCalledWith('/api/mock/conversion_hosts', [
      { mock: 'post body' }
    ]);
  });

  it('renders null when not posting, rejected or complete', () => {
    const component = shallow(<ConversionHostWizardResultsStep {...getBaseProps()} />);
    expect(component.get(0)).toBeFalsy();
  });

  it('renders a loading state when posting conversion hosts', () => {
    const component = shallow(<ConversionHostWizardResultsStep {...getBaseProps()} isPostingConversionHosts />);
    expect(component).toMatchSnapshot();
  });

  it('renders an error state when post was rejected', () => {
    const component = shallow(<ConversionHostWizardResultsStep {...getBaseProps()} isRejectedPostingConversionHosts />);
    expect(component).toMatchSnapshot();
  });

  it('renders a success state when there are post results', () => {
    const component = shallow(
      <ConversionHostWizardResultsStep {...getBaseProps()} postConversionHostsResults={[{ mock: 'results' }]} />
    );
    expect(component).toMatchSnapshot();
  });
});
