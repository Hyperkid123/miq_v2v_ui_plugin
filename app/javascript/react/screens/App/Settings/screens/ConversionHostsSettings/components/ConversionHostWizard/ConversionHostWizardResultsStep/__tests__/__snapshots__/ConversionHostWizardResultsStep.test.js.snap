// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion host wizard results step renders a loading state when posting conversion hosts 1`] = `
<WizardLoadingState
  message=""
  title="Starting Conversion Host Configuration..."
/>
`;

exports[`conversion host wizard results step renders a success state when there are post results 1`] = `
<div
  className="wizard-pf-complete blank-slate-pf"
>
  <div
    className="modal-wizard-results-grey-icon"
  >
    <span
      className="fa fa-clock-o"
    />
  </div>
  <h3
    className="blank-slate-pf-main-action"
  >
    Conversion host configuration is in progress.
  </h3>
  <p
    className="blank-slate-pf-secondary-action"
  >
    This may take a few minutes. Progress of the configuration will be shown on the Conversion Hosts page.
  </p>
</div>
`;

exports[`conversion host wizard results step renders an error state when post was rejected 1`] = `
<WizardErrorState
  message="The conversion host cannot be configured. Check your settings and try again."
  title="Error Starting Conversion Host Configuration"
/>
`;
